javascript/javascript2/week2/hyf-bay/JS/filter_products.js

```diff
@@ -0,0 +1,51 @@
+const searchInput = document.getElementById('search');
+
+searchInput.addEventListener('input', () => {
+  main.innerText = '';
+  renderProducts(
+    products.filter((product) =>
+      product.name.toLowerCase().includes(searchInput.value.toLowerCase())
+    )
+  );
+});
+
+const priceInput = document.getElementById('price');
+
+priceInput.addEventListener('input', () => {
+  main.innerText = '';
+  renderProducts(
+    products.filter((product) => product.price <= priceInput.value)
+  );
+  if (!priceInput.value) {
+    renderProducts(products);
+  }
+});
+
+const ratingInput = document.getElementById('rating');
+
+ratingInput.addEventListener('input', () => {
+  main.innerText = '';
+  renderProducts(
+    products.filter((product) => product.rating >= ratingInput.value)
+  );
+  if (!ratingInput.value) {
+    renderProducts(products);
+  }
+});
+
+const sort = document.getElementById('sort');
+sort.addEventListener('change', () => {
+  if (sort.value === 'top-rated') {
+    main.innerText = '';
+    renderProducts(products.sort((a, b) => b.rating - a.rating));
+  } else if (sort.value === 'alphabetical') {
+    main.innerText = '';
+    renderProducts(products.sort((a, b) => a.name.localeCompare(b.name)));
+  } else if (sort.value === 'price-up') {
+    main.innerText = '';
+    renderProducts(products.sort((a, b) => a.price - b.price));
+  } else if (sort.value === 'price-down') {
+    main.innerText = '';
+    renderProducts(products.sort((a, b) => b.price - a.price));
+  }
```

Comment:
Instead of writing the if- else inside the event handler function, use a separate function to do your if else check and call that function inside event handlers, making it easier to understand and reusable